extractor/src/test/java/com/example/extractor/MainTest.java

```diff
@@ -37,4 +43,51 @@ private boolean containsModule(Configuration config, String moduleName) {
 
         return false;
     }
-}
+
+    @Test
+    public void testGeneratedConfigMatchesExpected() throws Exception {
+        String exampleFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example1.java";
+        String templateFilePath = "src/main/resources/config-template.xml";
+        String outputFilePath = "src/test/resources/generated-config.xml";
+        String expectedFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/config-example1-expected.xml";
+
+        ConfigSerializer.serializeConfig(exampleFilePath, templateFilePath, outputFilePath);
+
+        String generatedContent = new String(Files.readAllBytes(Path.of(outputFilePath)));
+        String expectedContent = new String(Files.readAllBytes(Path.of(expectedFilePath)));
+
+        // Sort the properties in both the generated and expected content
+        String sortedGeneratedContent = sortProperties(generatedContent);
+        String sortedExpectedContent = sortProperties(expectedContent);
+
+        assertEquals(sortedExpectedContent.trim(), sortedGeneratedContent.trim(), "The generated configuration does not match the expected output.");    }
+
+    private String sortProperties(String configContent) {
```

Comment:
This method should be at ConfigSerializer.
Junit method should just compare result file with expected file.

Reply:
How about i move it to TemplateProcessor and just have ConfigSerializer doing serialization. And, then calling TemplateProcessor's methods to build properties and then replace. 

Or, shall` buildProperties` be a part of ConfigSerializer?

Reply:
>buildProperties be a part of ConfigSerializer?

Yes, it should same to string value whole module or modules.

Reply:
Done